umaserverdemo/src/main/kotlin/com/lightspark/ReceivingVasp.kt

```diff
@@ -46,15 +64,128 @@ class ReceivingVasp(
     private val nonceCache = InMemoryNonceCache(Clock.System.now().epochSeconds)
     private val coroutineScope = CoroutineScope(Dispatchers.IO)
     private lateinit var senderUmaVersion: String
+    private val httpClient = HttpClient {
+        install(ContentNegotiation) {
+            json(
+                Json {
+                    isLenient = true
+                },
+            )
+        }
+    }
 
     suspend fun createInvoice(call: ApplicationCall): String {
+        val (status, data) = createUmaInvoice(call)
+        if (status != HttpStatusCode.OK) {
+            call.respond(status, data)
+            return data
+        } else {
+            call.respond(data)
+        }
         return "OK"
     }
 
     suspend fun createAndSendInvoice(call: ApplicationCall): String {
+        val senderUma = call.parameters["senderUma"] ?: run {
+            call.respond(HttpStatusCode.BadRequest, "SenderUma not provided.")
+            return "SenderUma not provided."
+        }
+        val (status, data) = createUmaInvoice(call, senderUma)
+        if (status != HttpStatusCode.OK) {
+            call.respond(status, data)
+            return data
+        }
+        val senderComponents = senderUma.split("@")
+        val sendingVaspDomain = senderComponents.getOrNull(1) ?: run {
+            call.respond(HttpStatusCode.BadRequest, "Invalid senderUma.")
+            return "Invalid senderUma."
+        }
+        val wellKnownConfiguration = "http://$sendingVaspDomain/.well-known/uma-configuration"
+        val umaEndpoint = try {
+            val response = httpClient.get(wellKnownConfiguration)
+            if (response.status != HttpStatusCode.OK) {
+                call.respond(
+                    HttpStatusCode.FailedDependency,
+                    "failed to fetch request / pay endpoint at $wellKnownConfiguration"
+                )
+                return "failed to fetch request / pay endpoint at $wellKnownConfiguration"
+            } else {
+                Json.decodeFromString<JsonObject>(
+                    response.bodyAsText())["uma_request_endpoint"]?.jsonPrimitive?.content ?: ""
```

Comment:
If the uma_request_endpoint is empty (in the `?:` case), we should fail, right?

Reply:
agreed